From 15bb8a85694c09c0987701239dfeb11e9e98dd5b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Wed, 2 Sep 2026 20:04:10 +0200
Subject: [PATCH] less: Support the -T option

This option is passed by mandoc(1), while we don't implement the
actual functionality here, this patch allows using tagfile support
with mandoc(1) when a more feature-full less(1) implementation is
installed.

Otherwise, we would have to detect support for -T option in the
pager at run-time in mandoc(1), which would occur run-time costs.
Note that other option (-s and -X) are already ignored by BusyBox
less(1), so this seems like a good solution for mandoc(1) compat.
---
 miscutils/less.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/miscutils/less.c b/miscutils/less.c
index 87f44a324..71c657bb8 100644
--- a/miscutils/less.c
+++ b/miscutils/less.c
@@ -1838,6 +1838,7 @@ int less_main(int argc, char **argv) MAIN_EXTERNALLY_VISIBLE;
 int less_main(int argc, char **argv)
 {
 	char *tty_name;
+	char *less_tagfile;
 	int tty_fd;
 
 	INIT_G();
@@ -1851,8 +1852,11 @@ int less_main(int argc, char **argv)
 	getopt32(argv, "EMmN~IF"
 		IF_FEATURE_LESS_TRUNCATE("S")
 		IF_FEATURE_LESS_RAW("R")
-		/*ignored:*/"sX"
+		/*ignored, for compatibility with mandoc*/"T:"
+		/*ignored:*/"sX",
+		&less_tagfile
 	);
+	(void)less_tagfile; /* suppress unused variable warning */
 	argv += optind;
 	num_files = argc - optind;
 	files = argv;
